### PR TITLE
feat: ai generated job description

### PR DESCRIPTION
**Pull request by muhammad-ammar**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [ ] [Version](https://github.com/edx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/edx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: taxonomy/models.py [hunk 1]
```diff
@@ -458,6 +460,7 @@ class Job(TimeStampedModel):
             'The title of job.'
         )
     )
+    description = models.TextField(default='', help_text='AI generated job description.')
```
Comment (sameenfatima78) [inline taxonomy/models.py:463]: shouldn't this field have a max character length? Unsure how much long texts can be indexed into Algolia.

File: taxonomy/tasks.py [hunk 2]
```diff
@@ -105,3 +105,17 @@ def update_xblock_skills_verification_counts(xblock_uuid, verified_skills, ignor
     LOGGER.info('[TAXONOMY] update_xblock_skills_verification_counts task triggered')
     utils.update_xblock_skills_verification_counts(xblock_uuid, verified_skills, ignored_skills)
     LOGGER.info('[TAXONOMY] update_xblock_skills_verification_counts task completed')
+
+
+@shared_task()
+def generate_job_description_task(external_id, name):
```
Comment (sameenfatima78) [inline taxonomy/tasks.py:111]: nit: remove task from the end

File: taxonomy/tasks.py [hunk 3]
```diff
@@ -105,3 +105,17 @@ def update_xblock_skills_verification_counts(xblock_uuid, verified_skills, ignor
...
+    """
+    Task to gernate and store job description.
```
Comment (sameenfatima78) [inline taxonomy/tasks.py:113]: nit: generate*

File: taxonomy/utils.py [hunk 4]
```diff
@@ -688,3 +695,41 @@ def update_xblock_skills_verification_counts(usage_key: str, verified_skills: Li
...
+    """
+    Generate and store a job description.
+
+    Arguments:
+        job_external_id (str): Lightcast job id
+        job_name (str): Job name
+    """
+    prompt = settings.JOB_DESCRIPTION_PROMPT.format(job_name=job_name)
+    description = chat_completion(prompt)
+    Job.objects.filter(external_id=job_external_id).update(description=description)
```
Comment (sameenfatima78) [inline taxonomy/utils.py:710]: This expression gives an impression that we are saving same description for multiple jobs. Can we use save() instead?

**Review by sameenfatima78 — APPROVED** [review]
some minor comments but looks good overall.

File: taxonomy/models.py (hunk 1 above)
Comment (muhammad-ammar) [reply taxonomy/models.py:463]: This will be small text. Around 50 words

File: taxonomy/utils.py (hunk 4 above)
Comment (muhammad-ammar) [reply taxonomy/utils.py:710]: I think `update` will be more performant because `filter` and `update` will happen in a single call.

**Review by saleem-latif — APPROVED** [review]

**Pull request merged**